xbmc/threads/platform/pthreads/ThreadImpl.cpp

```diff
@@ -116,12 +116,20 @@ void CThread::SetThreadInfo()
 
 ThreadIdentifier CThread::GetCurrentThreadId()
 {
+#if defined(TARGET_ANDROID)
+  return gettid();
+#else
   return pthread_self();
+#endif
 }
 
 bool CThread::IsCurrentThread(const ThreadIdentifier tid)
 {
+#if defined(TARGET_ANDROID)
+  return gettid() == tid;
```

Comment:
https://forum.kodi.tv/showthread.php?tid=339983 <- any chance this might go wrong?

Reply:
Refering to the logs it doesn't seem to be an issue caused by this PR, the thread ids are increasing, no resage there. So I would say no